### PR TITLE
DSD-1114: Button sizes

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Fixes JIRA ticket [DSD-1114](https://jira.nypl.org/browse/DSD-1114)

## This PR does the following:

- Adds new semantic tokens for spacing and sizes.
- Adds a new `size` prop to the `Button` component. The accepted values are a string literal type of "small", "medium" and "large".
- Updates `Header`, `Notification`, `SearchBar`, and `Tabs` which internally use the `Button` component with slight style tweaks. The `Header` had the most updates since its inherently custom.

## How has this been tested?

Locally in Storybook.

## Accessibility concerns or updates

- The `Button`'s accessibility section has been updated with W3C size information, along with our own recommendation on using the "medium" and "large" sizes and only the "small" if it's really needed.

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] Review the Vercel preview deployment once it is ready.

### CODE REVIEW

File: src/components/Notification/Notification.stories.mdx
Comment (bigfishdesign13) [inline src/components/Notification/Notification.stories.mdx:52]: Is this necessary simply because `Notification` uses `Button` internally?

File: src/components/SearchBar/SearchBar.tsx
Comment (bigfishdesign13) [inline src/components/SearchBar/SearchBar.tsx:173]: Why the switch to `sx`?

File: src/theme/components/button.ts
Comment (bigfishdesign13) [inline src/theme/components/button.ts:15]: Did the `minHeight` attribute get removed altogether?  This is necessary for the `medium` size in order to keep it’s height in sync with the height of `TextInput` and `Select`.

File: src/theme/components/button.ts
Comment (bigfishdesign13) [inline src/theme/components/button.ts:12]: There are some odd things happening with spacing and icon align.  Let’s take this opportunity to change this to `lineHeight: “1”`.  That should help.

File: src/components/Tabs/Tabs.stories.mdx
Comment (bigfishdesign13) [inline src/components/Tabs/Tabs.stories.mdx:44]: Is this necessary simply because `Tabs` uses `Button` internally?

**Review by bigfishdesign13 — CHANGES_REQUESTED** [review]
Overall, looks good, but there are some styling issues that should be addressed.

File: src/components/Notification/Notification.stories.mdx
Comment (EdwinGuzman) [reply src/components/Notification/Notification.stories.mdx:52]: Yes.

File: src/components/SearchBar/SearchBar.tsx
Comment (EdwinGuzman) [reply src/components/SearchBar/SearchBar.tsx:173]: Styles didn't propagate down correctly.

File: src/components/Tabs/Tabs.stories.mdx
Comment (EdwinGuzman) [reply src/components/Tabs/Tabs.stories.mdx:44]: Yep.

File: src/theme/components/button.ts
Comment (EdwinGuzman) [reply src/theme/components/button.ts:15]: The `minHeight` doesn't allow the padding for the medium or large size to display correctly. I was only able to get the right size by removing it.

Comment (EdwinGuzman) [reply src/theme/components/button.ts:15]: Added it back.

File: src/theme/components/button.ts
Comment (EdwinGuzman) [reply src/theme/components/button.ts:12]: Added.

File: src/components/StyleGuide/Buttons.stories.mdx
Comment (EdwinGuzman) [inline src/components/StyleGuide/Buttons.stories.mdx]: This should not be "TBD" since this is the default. So maybe we can say that this covers almost all use cases whenever a button is needed.

**Comment by EdwinGuzman** [comment]
I had trouble with the `Pill` variant because the height for the default size is not always consistent. In the following screenshot, the top part is from the styleguide and the bottom part is from the doc page. The size is set to the default "medium" but the `Pill` is visibly different.

Screen Shot 2022-09-26 at 3 02 53 PM

File: src/components/StyleGuide/Buttons.stories.mdx
Comment (bigfishdesign13) [reply src/components/StyleGuide/Buttons.stories.mdx]: Thanks for calling me out.  I was being lazy.  This has been addressed.

**Comment by bigfishdesign13** [comment]
> I had trouble with the `Pill` variant because the height for the default size is not always consistent. In the following screenshot, the top part is from the styleguide and the bottom part is from the doc page. The size is set to the default "medium" but the `Pill` is visibly different.

@EdwinGuzman  Good catch.  I had made an adjustment to address the height for the other `pill` sizes, but I missed the change that would also fix this for the `medium` size.  This has now been fixed.

**Comment by EdwinGuzman** [comment]
The last update looks good. Let's get another set of eyes on this before merging.

**Comment by bigfishdesign13** [comment]
@oliviawongnyc, @jackiequach and @Toxiapo:
Can at least one of you review this PR today or early tomorrow?  Thanks.

File: CHANGELOG.md
Comment (jackiequach) [inline CHANGELOG.md]: Missing "- Updates..."

File: src/components/Button/Button.stories.mdx
Comment (jackiequach) [inline src/components/Button/Button.stories.mdx]: put prop values in back ticks for consistency with the component props , i.e. `"small"`

File: src/components/Button/Button.stories.mdx
Comment (jackiequach) [inline src/components/Button/Button.stories.mdx]: back ticks here

File: src/components/StyleGuide/Buttons.stories.mdx
Comment (jackiequach) [inline src/components/StyleGuide/Buttons.stories.mdx]: back ticks

**Review by jackiequach — COMMENTED** [review]
Looks good! Just some minor comments on documentation.

File: CHANGELOG.md
Comment (oliviawongnyc) [inline CHANGELOG.md:24]: Comment appears to be unfinished.

**Review by oliviawongnyc — APPROVED** [review]
Looks good to me! One small comment in CHANGELOG.md.

File: CHANGELOG.md
Comment (EdwinGuzman) [reply CHANGELOG.md]: This was also pointed out by @oliviawongnyc . I was trying to move two lines but just moved one instead. Good catch.

**Review by bigfishdesign13 — APPROVED** [review]
All looks good now.

**Review by Toxiapo — APPROVED** [review]
LGTM!

**Pull request merged**
